### PR TITLE
fix: rides history page

### PR DESCRIPTION
**Pull request by TomRytt**
# Description
Fixes issue #856 by adding the required "arrival_time_from" and "arrival_time_to" fields to the gtfs_ride_stops api call. 
Changed the previous api call which utilized the gtfsAPI to an axios call directly to the server with the required data since the gtfsAPI whitelisted the required fields mentioned above which caused the api call to fail.

Still requires futher optimizations since the request is very slow and tests need to be written in future issues.

@NoamGaash 

image
image

### CODE REVIEW

File: src/api/gtfsService.ts [hunk 1]
```diff
@@ -118,9 +126,33 @@ export async function getGtfsStopHitTimesAsync(stop: BusStop, timestamp: Moment)
...
 
   const rideIds = closestInTimeRides.map((ride) => ride.id).join(JOIN_SEPARATOR)
-  const stopHits = await GTFS_API.gtfsRideStopsListGet({
-    gtfsRideIds: rideIds,
-    gtfsStopIds: stop.stopId.toString(),
-  })
-  return stopHits.sort((hit1, hit2) => +hit1.arrivalTime! - +hit2.arrivalTime!)
+
+  const maxEndTime = Math.max(...rides.map((ride) => Number(ride.endTime)))
+  const minStartTime = Math.min(...rides.map((ride) => Number(ride.startTime)))
```
Comment (NoamGaash) [inline src/api/gtfsService.ts:131]: just an idea - sometimes the actual ride happen before / after this range. maybe we should consider something like -
```suggestion
  const timeMargin = 1000 * 60 * 60 * 1.5 // 1.5 hours
  const maxEndTime = Math.max(...rides.map((ride) => Number(ride.endTime))) + timeMargin
  const minStartTime = Math.min(...rides.map((ride) => Number(ride.startTime))) + timeMargin
```

**Review by NoamGaash — APPROVED** [review]
looks amazing :clap: 
would you like some help with the linter?

File: src/api/gtfsService.ts (hunk 1 above)
Comment (TomRytt) [reply src/api/gtfsService.ts]: Thanks for the quick response. I fixed the linter issues, added the correct type and added error & empty data error "handling" (can be improved) and submitted before seeing your replies. Would you like me to incorporate your suggestion and create another commit?

File: src/api/gtfsService.ts [hunk 2]
```diff
@@ -118,9 +130,41 @@ export async function getGtfsStopHitTimesAsync(stop: BusStop, timestamp: Moment)
...
+
+    if (stopHitsRes.data.length === 0) {
+      throw new Error(`No stop hits found`)
+    }
+
+    const stopHits: GtfsRideStopPydanticModel[] = stopHitsRes.data
+
+    return stopHits.sort((hit1, hit2) => +hit1.arrivalTime! - +hit2.arrivalTime!)
+  } catch (error) {
+    console.error(`Error fetching stop hits:`, error)
```
Comment (NoamGaash) [inline src/api/gtfsService.ts:167]: nice. I guess we can use some error boundary component to render a meaning full error message to the user

File: src/api/gtfsService.ts (hunk 1 above)
Comment (NoamGaash) [reply src/api/gtfsService.ts]: It's up to you, it was just a suggestion.
Thanks for fixing the lint issues!

**Review by NoamGaash — APPROVED** [review]

**Comment by NoamGaash** [comment]
Thank you!
@all-contributors please add @TomRytt  as code contributor

**Pull request merged**